docs/developers/default-aliases.md

```diff
@@ -0,0 +1,30 @@
+# Server Based Default Aliases
+
+Some aliases are common, so it makes some sense to predefine them on the server, rather than adding them to all incoming requests (or all stored requests). We can define the aliases in a file to be read at server startup and injected into all requests.
```

Comment:
So... I'm a bit puzzled by some of the design decisions here. Broadly speaking, I have two independent concerns:

------------------------------------

For features like these, there are two key questions to ask:

1. Should the values vary from deployment to deployment?
2. Will we need to change these values on-the-fly, without doing a release?

Based on these, we can categorize the type of values we're working with:

If both are "no", it's a **Constant**.
If only (1) is "yes", it's **Configuration**.
If both are "yes," it's **Data**. (I don't know if there's a standard name for this case... but it's what you'd put in a database, or fetch from another Service via HTTP, or whatever) 

**Constants** can be hardcoded... but I think everyone agrees that this feature doesn't fit that category.

**Configuration** in modern apps is done via environment variables. See [the 12 factor app docs](https://12factor.net/config).

We never discussed whether this should be **Data**... but I can see arguments in either direction. It doesn't matter much to me either way.

What I do think does matter, though, is that we use the same tools to accomplish the same goals.

If Config: this feature shouldn't force people to read config from files. Even though environment variables are ideal, we do use Viper to accommodate legacy infrastructures because PBS is open-source. If these values are config, they should come straight through Viper too.

If Data: Consider that people will want to use different backends, just like they do with Stored Requests. It's fine if "Files" are the only backend you want to implement now... but let's choose config values which can scale to other backends without breaking changes. Alternatively: you could probably extend today's Stored Request backends to add support here.

------------------------------------------------------------------

The other issue on my mind is Parsimony (see [this link](https://engineering.squarespace.com/blog/2018/six-attributes-of-beautiful-systems)). We have a few other use-cases for "global" Request values today.

1. `request.at` (signals first-price or second-price auction. In header bidding, "first price" makes a good default).
2. `request.tmax` (currently `auction_timeouts_ms.default` in the Viper config)
3. `regs.ext.gdpr` (currently `gdpr.usersync_if_ambiguous` in the Viper config)

All these features, as well as this new one you're adding, could be the same feature if you just merge the JSON in at the top level.

I'm not suggesting you tackle all that in this PR... but if you agree, it would make sense to implement this feature as a top-level Request merge. Then later we can deprecate/delete the other config flags, which will make PBS overall more Parsimonious.